workloads/kube-burner/common.sh

```diff
@@ -12,21 +17,34 @@ fi
 export TOLERATIONS="[{key: role, value: workload, effect: NoSchedule}]"
 export UUID=$(uuidgen)
 
-log() {
-  echo -e "\033[1m$(date "+%d-%m-%YT%H:%M:%S") ${@}\033[0m"
-}
+# Check if we're on bareMetal
+export baremetalCheck=$(oc get infrastructure cluster -o json | jq .spec.platformSpec.type)
+
+#Check to see if the infrastructure type is baremetal to adjust script as necessary 
+if [[ "${baremetalCheck}" == '"BareMetal"' ]]; then
+  log "BareMetal infastructure: setting isBareMetal accordingly"
+  export isBareMetal=true
+else
+  export isBareMetal=false
+fi
+
 
 deploy_operator() {
-  log "Removing benchmark-operator namespace, if it already exists"
-  oc delete namespace benchmark-operator --ignore-not-found
-  log "Cloning benchmark-operator from branch ${OPERATOR_BRANCH} of ${OPERATOR_REPO}"
-  rm -rf benchmark-operator
-  git clone --single-branch --branch ${OPERATOR_BRANCH} ${OPERATOR_REPO} --depth 1
-  (cd benchmark-operator && make deploy)
-  kubectl apply -f benchmark-operator/resources/backpack_role.yaml
-  kubectl apply -f benchmark-operator/resources/kube-burner-role.yml
-  log "Waiting for benchmark-operator to be running"
-  oc wait --for=condition=available "deployment/benchmark-controller-manager" -n benchmark-operator --timeout=300s
+  if [[ "${isBareMetal}" == "false" ]]; then
```

Comment:
tbh i think we should just leave the operator around each time and change this to apply to all cluster types. I don't see a reason why we need to delete/recreate the operator

Reply:
++

Reply:
Once the operator is deployed, we shouldn't delete it, I'd modify the current code to check if operator is running, if so, do nothing, otherwise deploy it.

Reply:
Removed delete command